### PR TITLE
rest api

### PR DESCRIPTION
**Pull request by egabancho**

### CODE REVIEW

File: invenio_sequencegenerator/views.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+from .permissions import need_permission
+
+blueprint = Blueprint(
+    'invenio_sequencegenerator',
+    __name__,
+    template_folder='templates'
+)
+
+
+def pass_keywords(f):
```
Comment (zzacharo) [inline invenio_sequencegenerator/views.py:27]: I am not sure if we gain much at least in the current state with this :) Except if you have in mind to enrich it in the future..

File: invenio_sequencegenerator/utils.py [hunk 2]
```diff
@@ -1,17 +1,45 @@
...
 """Utilities for template strings."""
 
 from __future__ import absolute_import, print_function
 
 import re
 
+from flask import current_app
+from werkzeug.utils import import_string
+
+from ._compat import string_types
```
Comment (kpsherva) [inline invenio_sequencegenerator/utils.py:17]: I think you could use [six](https://pythonhosted.org/six/#six.string_types)

File: invenio_sequencegenerator/permissions.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+def allow_all(*args, **kwargs):
+    """Return permission that always allow an access.
+
+    :returns: A object instance with a ``can()`` method.
+    """
+    return type('Allow', (), {'can': lambda self: True})()
+
+
+def defautl_permission_factory(action, template, keywords):
```
Comment (kpsherva) [inline invenio_sequencegenerator/permissions.py]: minor: typo `default_permission_factory`

File: invenio_sequencegenerator/utils.py (hunk 2 above)
Comment (egabancho) [reply invenio_sequencegenerator/utils.py:17]: You are definitely right, on the other hand I think it's not worth adding yet another dependency just to get the `string_types` right, I believe we use this same pattern in [some other places](https://github.com/search?q=org%3Ainveniosoftware+org%3Ainveniosoftware+_compat&type=Code) across Invenio.
What do you think?

File: invenio_sequencegenerator/views.py (hunk 1 above)
Comment (egabancho) [reply invenio_sequencegenerator/views.py:27]: > Except if you have in mind to enrich it in the future..

I started writing `keywords = request.get_json()` inside the `post` method, and I though "what if you'll do this somewhere else?" 😄 
Then, this came up to be handy when building the permissions, I believe for the CDS-Videos case we'll need to check the keywords, hence I think the decorator it's needed.

Comment (zzacharo) [reply invenio_sequencegenerator/views.py:27]: Ok I thought also you had in mind something else also :)

**Review by zzacharo — APPROVED** [review]

**Comment by egabancho** [comment]
No longer WIP, let me know if you both are OK with it and I'll merge it and release it ☺️

**Comment by zzacharo** [comment]
@egabancho Fine with me!

**Comment by egabancho** [comment]
No logger needed (for now).

**Pull request closed without merging**